app/assets/javascripts/discourse/app/components/topic-map/topic-map-summary.gjs

```diff
@@ -1,162 +1,443 @@
+  @tracked views = [];
+  @tracked loading = true;
+
+  get shouldShowParticipants() {
+    return (
+      this.args.topic.posts_count >= MIN_POSTS_COUNT &&
+      this.args.topicDetails.participants?.length >=
+        MIN_USERS_COUNT_FOR_AVATARS &&
+      !this.site.mobileView
+    );
   }
 
-  get createdByUsername() {
-    return this.args.topicDetails.created_by?.username;
+  get readTime() {
+    const calculatedTime = Math.ceil(
+      Math.max(
+        this.args.topic.word_count / this.siteSettings.read_time_word_count,
+        (this.args.topic.posts_count * MIN_POST_READ_TIME) / 60
+      )
+    );
+
+    return calculatedTime > 3 ? calculatedTime : null;
   }
 
-  get lastPosterUsername() {
-    return this.args.topicDetails.last_poster?.username;
+  get topRepliesSummaryEnabled() {
+    return this.args.postStream.summary;
   }
 
-  get toggleMapButton() {
-    return {
-      title: this.args.collapsed
-        ? "topic.expand_details"
-        : "topic.collapse_details",
-      icon: this.args.collapsed ? "chevron-down" : "chevron-up",
-      ariaExpanded: this.args.collapsed ? "false" : "true",
-      ariaControls: "topic-map-expanded__aria-controls",
-      action: this.args.toggleMap,
-    };
+  get topRepliesTitle() {
+    if (this.topRepliesSummaryEnabled) {
+      return;
+    }
+
+    return I18n.t("summary.short_title");
   }
 
-  get shouldShowParticipants() {
+  get topRepliesIcon() {
+    return this.topRepliesSummaryEnabled ? "arrows-alt-v" : "layer-group";
+  }
+
+  get topRepliesLabel() {
+    return this.topRepliesSummaryEnabled
+      ? I18n.t("summary.show_all_label")
+      : I18n.t("summary.short_label");
+  }
+
+  get loneStat() {
+    const hasViews = this.args.topic.views >= 0;
+    const hasLikes =
+      this.args.topic.like_count > MIN_LIKES_COUNT &&
+      this.args.topic.posts_count > MIN_POSTS_COUNT;
+    const hasLinks = this.linksCount > 0;
+    const hasUsers = this.args.topic.participant_count > MIN_PARTICIPANTS_COUNT;
+    const canSummarize = this.args.topic.has_summary;
+
+    if (canSummarize) {
+      return false;
+    }
+
     return (
-      this.args.collapsed &&
-      this.args.topic.posts_count > 2 &&
-      this.args.topicDetails.participants &&
-      this.args.topicDetails.participants.length > 0
+      [hasViews, hasLikes, hasLinks, hasUsers].filter(Boolean).length === 1
     );
   }
 
-  get createdByAvatar() {
-    return htmlSafe(
-      avatarImg({
-        avatarTemplate: this.args.topicDetails.created_by?.avatar_template,
-        size: "tiny",
-        title:
-          this.args.topicDetails.created_by?.name ||
-          this.args.topicDetails.created_by?.username,
-      })
+  get linksCount() {
+    return this.args.topicDetails.links?.length ?? 0;
+  }
+
+  get topicLinks() {
+    return this.args.topicDetails.links;
+  }
+
+  get linksToShow() {
+    return this.allLinksShown
+      ? this.topicLinks
+      : this.topicLinks.slice(0, TRUNCATED_LINKS_LIMIT);
```

Comment:
In `linksCount` above, you're using `.links?.length` so I'm assuming it might not always be present here either?

```suggestion
      : this.topicLinks?.slice(0, TRUNCATED_LINKS_LIMIT);
```